### ISSUE TITLE
pain.001;SPS;2022 standard validation fails

### ISSUE THREAD

**Issue by sdespont** [ch2877/swiss-payment#10]
SIX validation : https://validation.iso-payments.ch for the last standard "pain.001;SPS;2022" returns an error

Someone know if the XSD path must be updated?

```
(Line number) - Errors
******************************************************
(0002) - Error: no declaration found for element 'Document'


(0001) <?xml version="1.0" encoding="UTF-8"?>
(0002) <Document xmlns="http://www.six-interbank-clearing.com/de/pain.001.001.03.ch.02.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.six-interbank-clearing.com/de/pain.001.001.03.ch.02.xsd pain.001.001.03.ch.02.xsd">
```

**Comment by eric-reichenbach** [comment]
There's a lot of modifications in the new standard https://www.six-group.com/dam/download/banking-services/interbank-clearing/fr/standardization/iso/swiss-recommendations/implementation-guidelines-ct-developer_support-2022.pdf

As this new modifications are due on the **18th of november**, I'll look into it. We might need to be able to support both versions at the same time to ease the transition. I'm not sure which pattern would be the most efficient however. 

- We'll need new CreditTransfer classes for the renamed payment types.
- Some nodes have new names 
- Some validation rules have been altered

I wonder if we should do a new fork for the new Specs? It would allow to generate both versions in parallel to transition smoothly.

@toooni , @sprain , @a-schild

**Comment by sdespont** [comment]
Yes, this library must support both versions. In my point of view, `CustomerCreditTransfer` constructor must contains a new optional parameter permitting the switch with the new specs version 2022. That way, there is no BC and no need to create a new fork.

After reading the spec 2022, I think that the main changes are :
- New `Document` header with new XSD `<Document xmlns="urn:iso:std:iso:20022:tech:xsd:pain.001.001.09" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:iso:std:iso:20022:tech:xsd:pain.001.001.09 pain.001.001.09.ch.03.xsd">`
- `CtctDtls` drops `Nm` and add company and software details in `Othr`
- New way to format dates permitting to specify a "date" or "datetime"

**Comment by sdespont** [comment]
Also, the new version "could" (you know how banking institutions are up to date...) be accepted since end of November 2022, but the current one has a grace period of a year before being refused.

https://www.six-group.com/en/products-services/banking-services/standardization/iso-payments.html

![image](https://user-images.githubusercontent.com/646632/185873149-0ae6d67c-0716-4973-a491-cc439545587c.png)

**Comment by sprain** [comment]
> I wonder if we should do a new fork for the new Specs? It would allow to generate both versions in parallel to transition smoothly.

I'd suggest to handle it by releasing a new `v3.0` major version.
Thus makes upgrading to the new specs a deliberate move and at the same time leaves room for fixes in `v2.0.x`.

**Comment by will2877** [comment]
Are there any other changes that would need to be addressed in a `v3.0`

**Comment by sdespont** [comment]
Even if SPS 2021 is still valid until november 2025, it would be a good thing to address all needed changes before.

![image](https://github.com/user-attachments/assets/735b2270-4780-4a38-ac35-580384a29aeb)
